Resources/frontend/js/jquery.adyen-payment-selection.js

```diff
@@ -399,30 +390,37 @@
             if (!this.__hasActivePaymentMethod()) {
                 return false;
             }
-            var storedPaymentMethod =  this.sessionStorage.getItem(this.paymentMethodSession);
+            var storedPaymentMethod = this.sessionStorage.getItem(this.paymentMethodSession);
             var keys = Object.keys(storedPaymentMethod);
 
             return 1 === keys.length && 'type' === keys[0]; // Minimal state structure @see __buildMinimalState()
         },
         /**
-         * @param {string} paymentMethodType
+         * @param {object} paymentMethod
          * @return {boolean}
          * @private
          */
-        __isAdyenPaymentMethod: function (paymentMethodType) {
-            return paymentMethodType.indexOf(this.opts.adyenPaymentMethodPrefix) !== -1;
+        __isAdyenPaymentMethod: function (paymentMethod) {
+            return paymentMethod.isAdyenPaymentMethod || false;
         },
         /**
          * @param {object} paymentMethod
          * @return {boolean}
          * @private
          */
         __canHandlePayment: function (paymentMethod) {
-            if (this.__isStoredPaymentMethod(paymentMethod.id || '')) {
+            var me = this;
+
+            if (!me.__isAdyenPaymentMethod(paymentMethod))  {
+                return false;
+            }
+
+            if (this.__isStoredPaymentMethod(paymentMethod)) {
                 return true;
             }
 
-            return "undefined" !== typeof paymentMethod.details;
+            // not all adyen payment methods have "details", these cannot be handled by webcomponents (e.g. Paypal)
+            return "undefined" !== typeof paymentMethod.metadata.details;
```

Comment:
Reminder to test Blik with this, as it is a PM that doesn't have details

Reply:
@acampos1916 
We mostly use Paypal to test this flow, should we also add Blik to our testing or is PayPal enough?
Blik would be a bit more cumbersome due to the country restriction.